File: db/conference_image.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in conference_image.
*/

class Conference_Image extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "conference_image";
    $this->order = "";
    $this->domain = "conference";
    $this->field['conference_id']['type'] = 'INTEGER';
    $this->field['conference_id']['not_null'] = true;
    $this->field['mime_type_id']['type'] = 'INTEGER';
    $this->field['mime_type_id']['not_null'] = true;
    $this->field['image']['type'] = 'BYTEA';
    $this->field['image']['not_null'] = true;
    $this->field['conference_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/conference_image.rb
module Momomoto
  class Conference_image < Base
    def initialize
      super
      @domain = 'conference'
      @log_changes = true
      @fields = {
        :conference_id => Datatype::Integer.new( {:not_null=>true, :primary_key=>true} ),
        :mime_type_id => Datatype::Integer.new( {:not_null=>true} ),
        :image => Datatype::Bytea.new( {:not_null=>true} ),
        :last_modified => Datatype::Timestamp.new( {:with_timezone=>true, :not_null=>true, :default=>true, :auto_update=>true} ),
        :last_modified_by => Datatype::Integer.new( {} )
      }
    end
  end
end
